Description: disable *fghJoystickWalkUSBdev on kfreeBSD* systems. Prevents FTBFS
Author: Anton Gladky
Applied-Upstream: https://svn.redports.org/gahr/graphics/freeglut/freeglut-2.8.0.diff
Last-Update: 2012-06-22

--- a/src/freeglut_joystick.c
+++ b/src/freeglut_joystick.c
@@ -151,7 +151,11 @@
 #                include <libusbhid.h>
 #            endif
 #        endif
-#        include <legacy/dev/usb/usb.h>
+#        if !(defined(__FreeBSD__) || defined(__FreeBSD_kernel__))
+#            include <legacy/dev/usb/usb.h>
+#        else
+#            include <dev/usb/usb_ioctl.h>
+#        endif
 #        include <dev/usb/usbhid.h>
 
 /* Compatibility with older usb.h revisions */
@@ -198,6 +202,7 @@
  */
 static char *fghJoystickWalkUSBdev(int f, char *dev, char *out, int outlen)
 {
+# if !(defined(__FreeBSD__) || defined(__FreeBSD_kernel__))  
   struct usb_device_info di;
   int i, a;
   char *cp;
@@ -219,6 +224,7 @@
         return out;
       }
   }
+# endif
   return NULL;
 }
 
